inference-engine/src/gna_plugin/transformations/broadcast_const.cpp

```diff
@@ -0,0 +1,171 @@
+            const_out_shape[0] == 1 &&
+            const_out_shape[2] == 1 &&
+            const_out_shape[3] == 1);
+}
+#endif
+
+bool HasDynamicShape(Node node) {
+    const auto & shape = node->get_output_partial_shape(0);
+    return (shape.rank().is_dynamic() || shape[0].is_dynamic());
+}
+
+bool DoTransformation(Node const_node, Node eltwise_node) {
+    if (HasDynamicShape(const_node) || HasDynamicShape(eltwise_node))
+        return false;
+
+#ifdef GNA_LEGACY
+    /* That is work around the problem with the next transformations
+        ngraph::pass::ConvertOpSet1ToLegacy -> ngraph::pass::BiasFusions ->
+                                                    ngraph::pass::ConvAddFusion, ngraph::pass::ConvMultiplyFusion
+        That transormations fuse bias into convolution and recognizes const node as [1, C, 1, 1].
+        TODO: remove that work around after removing ConvertOpSet1ToLegacy transormations
+    */
+    if (HasConvolutionInput(eltwise_node) && IsFusedBiasConst(const_node))
+        return false;
+#endif
+    const ngraph::Shape & eltwise_out_shape = eltwise_node->get_output_tensor(0).get_shape();
+    const size_t eltwise_out_dims_product = ngraph::shape_size(eltwise_out_shape);
+
+    {
+        const ngraph::Shape & const_out_shape = const_node->get_output_tensor(0).get_shape();
+        const size_t const_out_dims_product = ngraph::shape_size(const_out_shape);
+
+        if (const_out_shape.size() > eltwise_out_shape.size() ||
+            const_out_dims_product == eltwise_out_dims_product ||
+            eltwise_out_dims_product % const_out_dims_product) {
+            return false;
+        }
+    }
+
+    auto new_const_node = CreateTiledConst(const_node, eltwise_out_shape);
+
+    if (!new_const_node)
+        return false;
+
+    ngraph::replace_node(const_node, new_const_node);
+
+    return true;
+}
+
+} // namespace
+
+/**
+ * @brief Cannot use precise ngraph pattern since we can have arbitary number of non-functional
+ * nodes between Const, FakeQuantize and Eltwise layers
+ */
+
+BroadcastConst::BroadcastConst() {
+    MATCHER_SCOPE(BroadcastConst);
+
+    auto constant = ngraph::pattern::wrap_type<ngraph::opset8::Constant>();
+    auto fake_quantize = ngraph::pattern::wrap_type<ngraph::opset8::FakeQuantize>({constant,
+        ngraph::pattern::wrap_type<ngraph::opset8::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset8::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset8::Constant>(),
+        ngraph::pattern::wrap_type<ngraph::opset8::Constant>()});
+    auto eltwise_input = std::make_shared<ngraph::pattern::op::Or>(ngraph::OutputVector{constant, fake_quantize});
+    auto eltwise = ngraph::pattern::wrap_type<ngraph::opset8::Add,
+                                    ngraph::opset8::Subtract,
+                                    ngraph::opset8::Multiply,
+                                    ngraph::op::Eltwise>({eltwise_input, ngraph::pattern::any_input()});
```

Comment:
does this matcher cover the case when a constant is the second input?

Reply:
Yes, It covers. I added unit test to check it